File: segundos 4 lenguajes/c#/ProyectoCsharp/JugadorMagico.cs
namespace ProyectoJugadores
{
    public class JugadorMagico
    {
        public string Nombre { get; set; }
        public int Nivel { get; set; }

        public JugadorMagico()
        {
            Nombre = "Mago";
            Nivel = 5;
        }

        public JugadorMagico(string nombre, int nivel)
        {
            Nombre = nombre;
            Nivel = nivel;
        }

        public static JugadorMagico Copy(JugadorMagico j)
        {
            return new JugadorMagico(j.Nombre, j.Nivel);
        }
    }
}

File: segundos 4 lenguajes/go/JugadorMagico.go
package main

type JugadorMagico struct {
	Nombre string
	Nivel  int
}

func NewJugadorMagico() JugadorMagico {
	return JugadorMagico{"Mago", 5}
}

func NewJugadorMagicoNombre(nombre string) JugadorMagico {
	return JugadorMagico{nombre, 10}
}

func NewJugadorMagicoParams(nombre string, nivel int) JugadorMagico {
	return JugadorMagico{nombre, nivel}
}

func NewJugadorMagicoCopy(j JugadorMagico) JugadorMagico {
	return JugadorMagico{j.Nombre, j.Nivel}
}

type AbstractMagico interface {
	LanzarHechizo()
}
